am-lib/src/main/java/uk/gov/hmcts/reform/amlib/AccessManagementService.java

```diff
@@ -45,21 +52,36 @@ public void createResourceAccess(ExplicitAccessRecord explicitAccessRecord) {
     }
 
     /**
-     * Returns `resourceJson` when record with userId and resourceId exist and has READ permissions, otherwise null.
+     * Returns FilterResourceResponse when record with userId and resourceId exist and has READ permissions,
+     * otherwise null.
      *
      * @param userId       (accessorId)
      * @param resourceId   resource id
      * @param resourceJson json
      * @return resourceJson or null
+     * @throws UnsupportedPermissionsException when permissions are above 31 or below 0.
      */
-    public JsonNode filterResource(String userId, String resourceId, JsonNode resourceJson) {
+    @SuppressWarnings("PMD")
```

Comment:
Let's remove suppression and let's use `ConcurrentHashMap` as advised. It makes sense to follow guidance here.

Reply:
Done

Reply:
👍